src/app/modules/submission/encrypt-submission/encrypt-submission.utils.ts

```diff
@@ -302,3 +303,33 @@ export const getPaymentAmount = (
     }
   }
 }
+
+/**
+ * Retrieves payment description by payment_type
+ *
+ * - `Fixed Payments` references the description to as legacy behaviour
+ * - `Variable Payments` references the name
+ * - `Products` references the product name and quantity separated by a comma
+ * @param form
+ * @param paymentProducts
+ */
+export const getPaymentIntentDescription = (
+  form: IPopulatedEncryptedForm,
+  paymentProducts?: StorageModeSubmissionContentDto['paymentProducts'],
+) => {
+  const formPaymentFields = form.payments_field
+  switch (formPaymentFields.payment_type) {
+    case PaymentType.Fixed:
+      // legacy behaviour of fixed payments where the product name is referred as description
+      return formPaymentFields.description
+    case PaymentType.Variable:
+      return formPaymentFields.name
+    case PaymentType.Products: {
+      if (!paymentProducts) return form.title
+      const productDescriptions = paymentProducts
+        .map((product) => `${product.data.name} x ${product.quantity}`)
+        .join(', ')
+      return productDescriptions
```

Comment:
Should we have a default case that returns an error?

Reply:
7e4084dfa